### PR TITLE
python312Packages.open-clip-torch: 2.24.0 -> 2.26.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

This PR fixes this currently broken package.

Changelog: https://github.com/mlfoundations/open_clip/releases/tag/v2.26.1

cc @iynaix

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Review by iynaix — APPROVED** [review]

File: pkgs/development/python-modules/open-clip-torch/default.nix [hunk 1]
```diff
@@ -1,43 +1,50 @@
...
     repo = "open_clip";
     rev = "refs/tags/v${version}";
-    hash = "sha256-ugbXnXiOY9FrNvr8ZxnAgZO/SLCVoXbRgupi8cUwflU=";
+    hash = "sha256-XjPOsGet8VNzwEwzz14f1nF3XOgpkb4OERIc6VrDDZ8=";
   };
 
-  nativeBuildInputs = [
+  build-system = [
+    pdm-backend
     setuptools
```
Comment (natsukium) [inline pkgs/development/python-modules/open-clip-torch/default.nix:44]:
```suggestion
```
I think we no longer need `setuptools`, either.

Comment (GaetanLepage) [reply pkgs/development/python-modules/open-clip-torch/default.nix]: Indeed, done.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337105` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.open-clip-torch</li>
    <li>python311Packages.open-clip-torch.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
  </ul>
</details>

**Pull request merged**